lib/core/src/display/rendering/scene.rs

```diff
@@ -1,24 +1,99 @@
-use crate::system::web::{get_element_by_id, dyn_into, Result};
+use wasm_bindgen::prelude::Closure;
+use crate::prelude::*;
+use crate::system::web::get_element_by_id;
+use crate::system::web::dyn_into;
+use crate::system::web::Result;
+use crate::system::web::StyleSetter;
+use crate::system::web::resize_observer::ResizeObserver;
 use web_sys::HtmlElement;
 use nalgebra::Vector2;
+use std::cell::RefCell;
+use std::rc::Rc;
+use std::fmt;
+
+// ======================
+// === ResizeCallback ===
+// ======================
+
+pub type ResizeCallback = Box<dyn Fn(&Vector2<f32>)>;
+
+// =================
+// === SceneData ===
+// =================
+
+#[derive(Derivative)]
+#[derivative(Debug)]
+struct SceneData {
+    dimensions : Vector2<f32>,
+    #[derivative(Debug="ignore")]
+    resize_callbacks : Vec<ResizeCallback>
+}
+
+impl SceneData {
+    pub fn new(dimensions : Vector2<f32>) -> Self {
+        let resize_callbacks = default();
+        Self { dimensions, resize_callbacks }
+    }
+}
+
+// =============
+// === Scene ===
+// =============
 
 /// A collection for holding 3D `Object`s.
-#[derive(Debug)]
+//#[derive(Debug)]
 pub struct Scene {
-    pub container : HtmlElement,
+    pub dom          : HtmlElement,
+    _resize_observer : ResizeObserver,
```

Comment:
Because `_resize_observer` will not be used, but we need to keep track of it until `Scene` is dropped. This functionality was moved to `GraphicsRenderer`.

Reply:
@wdanilo actually, I have been mentioning that DOM resizing was moved to `GraphicsRenderer`, but I made a separation of concerns and created `DomContainer` and `GraphicsRenderer` owns it. This is the right place to look it.